From 9929d7bbf961a67087b971d789096ca818898457 Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Thu, 1 Apr 2021 13:01:31 +0300
Subject: [PATCH 17/17] libubi: Rename libubi.h to libubi-fstools.h

Some builds may have libubi.h in the mtd-utils package.
To avoid conflicts, rename libubi.h in the fstools package.

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 CMakeLists.txt                        | 2 +-
 block.c                               | 2 +-
 libubi/{libubi.h => libubi-fstools.h} | 6 +++---
 libubi/libubi-tiny.h                  | 2 +-
 libubi/libubi.c                       | 2 +-
 libubi/ubiutils-common.c              | 2 +-
 6 files changed, 8 insertions(+), 8 deletions(-)
 rename libubi/{libubi.h => libubi-fstools.h} (99%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a0d2014..c9585b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ INCLUDE_DIRECTORIES(${ubox_include_dir})
 SET_TARGET_PROPERTIES(ubi-utils PROPERTIES COMPILE_FLAGS
 		"-ffunction-sections -fdata-sections")
 
-INSTALL(FILES libubi/libubi-tiny.h libubi/libubi.h libubi/ubi-media.h
+INSTALL(FILES libubi/libubi-tiny.h libubi/libubi-fstools.h libubi/ubi-media.h
 		libubi/ubi-user.h libubi/ubi-media.h
 	DESTINATION include
 )
diff --git a/block.c b/block.c
index c4228b4..75e4bb6 100644
--- a/block.c
+++ b/block.c
@@ -52,7 +52,7 @@
 #define AUTOFS_MOUNT_PATH       "/tmp/run/blockd/"
 
 #ifdef UBIFS_EXTROOT
-#include "libubi/libubi.h"
+#include "libubi/libubi-fstools.h"
 #endif
 
 enum {
diff --git a/libubi/libubi.h b/libubi/libubi-fstools.h
similarity index 99%
rename from libubi/libubi.h
rename to libubi/libubi-fstools.h
index fe2f7e0..8b091be 100644
--- a/libubi/libubi.h
+++ b/libubi/libubi-fstools.h
@@ -20,8 +20,8 @@
  * UBI (Unsorted Block Images) library.
  */
 
-#ifndef __LIBUBI_H__
-#define __LIBUBI_H__
+#ifndef __LIBUBI_FSTOOLS_H__
+#define __LIBUBI_FSTOOLS_H__
 
 #include <ctype.h>
 #include <stdlib.h>
@@ -496,4 +496,4 @@ int ubi_is_mapped(int fd, int lnum);
 }
 #endif
 
-#endif /* !__LIBUBI_H__ */
+#endif /* !__LIBUBI_FSTOOLS_H__ */
diff --git a/libubi/libubi-tiny.h b/libubi/libubi-tiny.h
index 0ede6cd..eaef403 100644
--- a/libubi/libubi-tiny.h
+++ b/libubi/libubi-tiny.h
@@ -14,7 +14,7 @@
 #ifndef _LIBUBI_TINY_H__
 #define _LIBUBI_TINY_H__
 
-#include "libubi.h"
+#include "libubi-fstools.h"
 
 int ubiattach(libubi_t libubi, char *mtd);
 int ubidetach(libubi_t libubi, char *mtd);
diff --git a/libubi/libubi.c b/libubi/libubi.c
index 4e970b3..b20ec3e 100644
--- a/libubi/libubi.c
+++ b/libubi/libubi.c
@@ -33,7 +33,7 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
-#include "libubi.h"
+#include "libubi-fstools.h"
 #include "libubi_int.h"
 
 /**
diff --git a/libubi/ubiutils-common.c b/libubi/ubiutils-common.c
index 2271927..06dd035 100644
--- a/libubi/ubiutils-common.c
+++ b/libubi/ubiutils-common.c
@@ -32,7 +32,7 @@
 #include <ctype.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include "libubi.h"
+#include "libubi-fstools.h"
 
 /**
  * get_multiplier - convert size specifier to an integer multiplier.
-- 
2.34.1.windows.1

